runtime/runtime-params-estimator/src/rocksdb.rs

```diff
@@ -0,0 +1,311 @@
+        );
+    }
+
+    let cost = gas_counter.elapsed();
+
+    if db_config.debug_rocksdb {
+        println!("# Cost: {:?}", cost);
+        print_levels_info(&db);
+    }
+
+    drop(db);
+    tmp_dir.close().expect("Could not clean up temp DB");
+
+    // Store generated input data in a file for reproducibility of any results
+    if db_config.pr_data_path.is_none() {
+        let mut stats_file = std::fs::OpenOptions::new()
+            .read(true)
+            .write(true)
+            .append(true)
+            .create(true)
+            .open("names-to-stats.txt")
+            .unwrap();
+        let stats_num = std::io::BufReader::new(&stats_file).lines().count();
+        let data_dump_path = format!("data_dump_{:<04}.bin", stats_num);
+
+        let mut dump = std::fs::File::create(&data_dump_path).unwrap();
+        dump.write_all(&data).unwrap();
+
+        writeln!(stats_file, "# DATA {} written to {}", stats_num, data_dump_path)
+            .expect("Writing to \"names-to-stats.txt\" failed");
+    }
+
+    cost
+}
+
+pub(crate) fn rocks_db_read_cost(config: &Config) -> GasCost {
+    let db_config = &config.rocksdb_test_config;
+    let tmp_dir = tempfile::TempDir::new().expect("Failed to create directory for temp DB");
+    let data = input_data(db_config, INPUT_DATA_BUFFER_SIZE);
+    let db = new_test_db(&tmp_dir, &data, &db_config);
+
+    if db_config.debug_rocksdb {
+        println!("# {:?}", db_config);
+        println!("# After setup / before measurement:");
+        print_levels_info(&db);
+    }
+
+    let mut prng: XorShiftRng = rand::SeedableRng::seed_from_u64(SETUP_PRANDOM_SEED);
+    let mut keys: Vec<usize> = (0..db_config.setup_insertions).map(|_| prng.gen()).collect();
+    if db_config.sequential_keys {
+        keys.sort();
+    } else {
+        // give it another shuffle to make lookup order different from insertion order
+        keys.shuffle(&mut prng);
+    }
+
+    let gas_counter = GasCost::measure(config.metric);
+
+    for i in 0..db_config.op_count {
+        let key = keys[i as usize % keys.len()];
+        db.get(&key.to_string()).unwrap();
+    }
+
+    let cost = gas_counter.elapsed();
+
+    if db_config.debug_rocksdb {
+        println!("# Cost: {:?}", cost);
+        print_levels_info(&db);
+    }
+
+    drop(db);
+    tmp_dir.close().expect("Could not clean up temp DB");
+
+    // Store generated input data in a file for reproducibility of any results
+    if db_config.pr_data_path.is_none() {
+        let mut stats_file = std::fs::OpenOptions::new()
+            .read(true)
+            .write(true)
+            .append(true)
+            .create(true)
+            .open("names-to-stats.txt")
+            .unwrap();
+        let stats_num = std::io::BufReader::new(&stats_file).lines().count();
+        let data_dump_path = format!("data_dump_{:<04}.bin", stats_num);
+
+        let mut dump = std::fs::File::create(&data_dump_path).unwrap();
+        dump.write_all(&data).unwrap();
```

Comment:
std::fs::write ?

Reply:
Right! I completely forgot that exists.
Also, while at it, I extracted some duplicated code into `fn backup_input_data(data: &[u8])`